lib/index.js

```diff
@@ -307,28 +307,25 @@ InsightAPI.prototype.blockEventHandler = function(hashBuffer) {
   }
 };
 InsightAPI.prototype.transactionEventHandler = function(txBuffer) {
-  var tx = new Transaction().fromBuffer(txBuffer);
-  var result = this.txController.transformInvTransaction(tx);
-
-  setTimeout(function(result, txBuffer) { // delay tx relay for 900ms to see if we receive transaction lock
-    var hash = dashcore.crypto.Hash.sha256sha256(txBuffer);
-    var id = hash.toString('binary');
-
-    if (this.node.services.dashd.zmqKnownTransactionLocks.get(id)) {
-      result.txlock = true;
-    }
+  const txVersion = txBuffer.slice(0, 1).toString('hex');
```

Comment:
Doesn't look like a scalable solution. Maybe we need to fix handlers in dashcore-node?
By the way, did we get the wrong data for 'tx' event too?

Reply:
I only got wrong data for txlock event. But because of time limits could only check like half a day. So couldn't exlude the possibility that it would occur there as well. Why do you think it is not scalable? It shouldn't take a lot of time to do this check.
Fixing it would be best, but looking at dashcore-node (https://github.com/dashevo/dashcore-node/blob/master/lib/services/dashd.js#L704-L718) I don't see how `message` could ever get overwritten, so I suspect the wrong data comes from the dashd daemon itself.

Reply:
@Cofresi Yeah, I did not find any points for that error too :(

I mentioned that it's not scalable coz when we introduce a new version of transactions this fix will not work and we will just ignore transactions.

Reply:
yes, that is much better. Thank you!